micro-rdk-server/esp32/main.rs

```diff
@@ -39,92 +44,80 @@ mod esp32 {
-        // then the entire provisioning step can be skipped
-        #[cfg(has_robot_config)]
-        {
-            use micro_rdk::common::credentials_storage::RobotConfigurationStorage;
-            use micro_rdk::common::credentials_storage::RobotCredentials;
-            use micro_rdk::common::credentials_storage::WifiCredentialStorage;
-            use micro_rdk::common::credentials_storage::WifiCredentials;
-
-            log::warn!("Unconditionally using build-time WiFi and robot configuration");
-
-            let storage = NVSStorage::new("nvs").unwrap();
-            log::info!("Storing static values from build time wifi configuration to NVS");
-            storage
-                .store_wifi_credentials(WifiCredentials::new(
-                    SSID.expect("[cfg(has_robot_config)]: missing WiFi SSID")
-                        .to_string(),
-                    PASS.expect("[cfg(has_robot_config)]: missing WiFi password")
-                        .to_string(),
-                ))
-                .expect("Failed to store WiFi credentials to NVS");
-
-            log::info!("Storing static values from build time robot configuration to NVS");
-            storage
-                .store_robot_credentials(
-                    RobotCredentials::new(
-                        ROBOT_ID
-                            .expect("[cfg(has_robot_config)]: missing robot id")
-                            .to_string(),
-                        ROBOT_SECRET
-                            .expect("[cfg(has_robot_config)]: missing robot secret")
-                            .to_string(),
-                    )
-                    .into(),
-                )
-                .expect("Failed to store robot credentials to NVS");
+        };
 
-            serve_web(None, repr, max_connection, storage);
-        }
-        #[cfg(not(has_robot_config))]
-        {
-            use micro_rdk::common::provisioning::server::ProvisioningInfo;
-            let mut info = ProvisioningInfo::default();
-            info.set_manufacturer("viam".to_owned());
-            info.set_model("test-esp32".to_owned());
-            let storage = NVSStorage::new("nvs").unwrap();
-            serve_web(Some(info), repr, max_connection, storage);
-        }
+        let mut info = ProvisioningInfo::default();
+        info.set_manufacturer("viam".to_owned());
+        info.set_model("test-esp32".to_owned());
 
-        #[cfg(feature = "qemu")]
-        let network = {
-            use micro_rdk::esp32::esp_idf_svc::hal::prelude::Peripherals;
-            log::info!("creating eth object");
-            let eth = EspEth::wrap(
-                EthDriver::new_openeth(Peripherals::take().unwrap().mac, sys_loop_stack.clone())
-                    .unwrap(),
-            )
-            .unwrap();
-            eth_configure(&sys_loop_stack, eth).unwrap()
-        };
+        serve_web(Some(info), repr, max_connections, storage);
```

Comment:
@acmorrow I left `serve_web` take an Option as input because `micro-rdk-ffi` still takes `Option<ProvisioningInfo>` in the `serve_` functions it uses. If we want to change it later, we can do it in a dedicated PR after these merge.